maven-embedder/src/test/java/org/apache/maven/cli/MavenCliTest.java

```diff
@@ -338,9 +338,9 @@ public void testMVNConfigurationFunkyArguments()
         String revision = System.getProperty( "revision" );
         assertEquals( "8.2.0", revision );
 
-        assertEquals( "bar ", request.getSystemProperties().getProperty( "foo" ) );
-        assertEquals( "bar two", request.getSystemProperties().getProperty( "foo2" ) );
-        assertEquals( "Apache Maven", request.getSystemProperties().getProperty( "label" ) );
+        assertEquals( "bar ", request.getUserProperties().getProperty( "foo" ) );
+        assertEquals( "bar two", request.getUserProperties().getProperty( "foo2" ) );
+        assertEquals( "Apache Maven", request.getUserProperties().getProperty( "label" ) );
```

Comment:
The system properties are still modified, right ?
https://github.com/apache/maven/blob/2a9f39336cec1d8e52d30cc48503d51ed8672536/maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java#L1787-L1793

Reply:
Yes, correct. This explicitly not subject of this PR. This statement should go away sometime in the future.